### PR TITLE
Infinispan Configuration enchancements

### LINKED ISSUE #25136: Support bouncyCastle libs  + Infinispan + FIPS
### Describe the bug

Infinispan: 12.1 / 14
Quarkus: 2.7.5.Final
Java 11 / 17

Seems that bouncy castle lib is not supported until Infinispan 14.0. But If I move on to infinispan 14 and configure Quarkus in order to use `BCFIPSJSSE` then I got the following error:

```
13:51:46,057 INFO  [app] Caused by: java.security.KeyStoreException: BCFKS not found
13:51:46,057 INFO  [app]        at java.base/java.security.KeyStore.getInstance(KeyStore.java:871)
13:51:46,057 INFO  [app]        at org.infinispan.commons.util.SslContextFactory.getTrustManagerFactory(SslContextFactory.java:172)
13:51:46,057 INFO  [app]        at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.initSsl(ChannelInitializer.java:154)
13:51:46,057 INFO  [app]        ... 20 more
13:51:46,057 INFO  [app] Caused by: java.security.NoSuchAlgorithmException: BCFKS KeyStore not available
13:51:46,057 INFO  [app]        at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:159)
13:51:46,057 INFO  [app]        at java.base/java.security.Security.getImpl(Security.java:701)
13:51:46,057 INFO  [app]        at java.base/java.security.KeyStore.getInstance(KeyStore.java:868)

```

My infisnispan configuration looks like the following one:
```
# Infinispan
quarkus.infinispan-client.trust-store=src/main/resources/server.jks
quarkus.infinispan-client.trust-store-password=password
quarkus.infinispan-client.trust-store-type=BCFKS

quarkus.infinispan-client.server-list=localhost:11222
quarkus.infinispan-client.use-auth=true
```

In my opinion, I am missing some configurations that we already have in orders [modules](https://quarkus.io/guides/security-customization#bouncy-castle-jsse-fips), something like `quarkus.infinispan-client.trust-store-provider=BCFIPS`

[Reproducer](https://github.com/pjgg/quarkus-test-suite/blob/feat/infisnipan_fips_reproducer/messaging/infinispan-grpc-kafka/src/test/java/io/quarkus/qe/messaging/infinispan/InfinispanKafkaIT.java)
```
git clone git@github.com:quarkus-qe/quarkus-test-suite.git
git checkout feat/infisnipan_fips_reproducer
```

Scenario
```
mvn clean verify -Dall-modules -pl messaging/infinispan-grpc-kafka -Dit.test=InfinispanKafkaIT
```

I am not sure if `BCFIPS` is supported and if does then, I am not sure how to use it. Should I add `quarkus-security` extension ?, and If I am able to set `BCFIPS` as a provider I understand that then later I will not need any [JVM security custom](https://docs.oracle.com/middleware/11119/wls/SECMG/fips.htm#A1233313) configuration, isn't it? (on `quarkus.http.ssl` is not need it)

### PR DESCRIPTION
**Pull request by karesti**
* Deprecate serverList for hosts (consistency with other extensions)
* Deprecate authUsername for username
* Deprecate authPassowrd for password
* Provide uri connection configuration
* Provide ssl provider, protocol and ciphers closes #25136

### CODE REVIEW

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java [hunk 1]
```diff
@@ -21,10 +21,31 @@
...
+    public Optional<String> hosts;
+
+    // @formatter:off
     /**
      * Sets the host name/port to connect to. Each one is separated by a semicolon (eg. host1:11222;host2:11222).
+     * @deprecated {@link #hosts} should be used to configure the list or uri for an uri connection string.
      */
+    // @formatter:on
     @ConfigItem
+    @Deprecated
```
Comment (geoand) [inline extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:48]: We should mention which property replaces it

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java [hunk 2]
```diff
@@ -66,12 +87,30 @@ public class InfinispanClientRuntimeConfig {
...
     @ConfigItem
+    Optional<String> username;
+
+    /**
+     * Sets username used by authentication.
+     *
+     * @deprecated {@link #username} should be used to configure the credentials username.
+     */
+    @ConfigItem
+    @Deprecated
```
Comment (geoand) [inline extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:98]: Same as above

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java [hunk 3]
```diff
@@ -66,12 +87,30 @@ public class InfinispanClientRuntimeConfig {
...
+    @ConfigItem
+    Optional<String> password;
+
     /**
      * Sets password used by authentication
+     *
+     * @deprecated {@link #password} should be used to configure the credentials password.
      */
     @ConfigItem
+    @Deprecated
```
Comment (geoand) [inline extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:113]: And here

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java (hunk 1 above)
Comment (karesti) [reply extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:48]: I'm mentioning that in the docs above.

Comment (geoand) [reply extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:49]: Sure, but anyone reading the Javadoc needs to look at another source of information to find it - that is not a great user experience

Comment (karesti) [reply extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:49]: I'm adding the tags in the javadocs too...

Comment (geoand) [reply extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:49]: Great, I see

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java (hunk 2 above)
Comment (karesti) [reply extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:99]: I added the javadoc deprecated these ones

**Comment by karesti** [comment]
@geoand I fixed the javadoc issue I missed

**Review by geoand — APPROVED** [review]
LGTM

**Comment by karesti** [comment]
is there anything else to be validated at this point @geoand ?

**Pull request merged**